test/index.ts

```diff
@@ -0,0 +1,115 @@
+import tape from "fresh-tape";
+
+import { Member } from "../src/member";
+
+const EXAMPLE_ID = "example";
+
+tape("Able to initialize a member", (t) => {
+  const member = new Member();
+
+  member.processFeeds();
+
+  t.pass("Able to process feeds with zero data");
+
+  t.end();
+});
+
+tape("Able to add a member by ID", (t) => {
+  const member = new Member();
+
+  const req = member.requestAdd(EXAMPLE_ID);
+
+  t.ok(req, "Generated a request");
+
+  const res = member.acceptRequest(req);
+
+  t.ok(res, "Generated response");
+
+  member.processFeeds();
+
+  t.pass("Processed feeds");
+
+  const expectedMembers = [member.id, EXAMPLE_ID];
+  t.deepEqual(
+    member.knownMembers,
+    expectedMembers,
+    "New member got added to list"
+  );
+
+  t.end();
+});
+
+tape("Able to add a member by ID and sync", (t) => {
+  const member = new Member();
+  const other = new Member({ initiator: member.id });
```

Comment:
This adds the requirement that the other member knows the initiator at start. QR code for `member` has to be given to `other` and for `requestAdd` the `other`'s QR-code has to be given to `member`. This can be reduced to only one member if the initiator isn't needed.

Reply:
My assumption was that only known members were allowed to request an add, therefore if a new peer was to be added, they'd need to know the ID of the first member in order to traverse the history and get to the latest state.

Ties into what we talked about regarding using just member feeds for representing history and traversing them instead of a group feed.

Reply:
> need to know the ID of the first member

The ID of the first member should be in the first _(and only?)_ block of the group feed, no?

But beside that: The workflow needs indeed to be `group-id` → new member → `new-member-id` → existing member.

Reply:
Hmm, yeah, my current code doesn't account for the group feed or how the initial member is found. Was hoping to punt that for later and focus on the request flow.

Reply:
It is okay to make this a memo.